### PR TITLE
1442/listing finder q3

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #1442 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR builds out q3 of the listings finder. Since q3 introduced a new pattern, it warranted a refactor of the finder to utilize separate custom components rather than trying to contain all the cases within finder.tsx. 

## How Can This Be Tested/Reviewed?

This can be tested by clicking through the form and ensuring that the form saves and applies the rental cost information entered whether you click previous on the disclaimer page or finish and see it reflected in the listings.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/src/forms/finder/FinderRentalCosts.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { FrontendListingFilterStateKeys } from "@bloom-housing/shared-helpers"
+import { Field, t } from "@bloom-housing/ui-components"
+import { UseFormMethods } from "react-hook-form"
+import { FinderQuestion } from "../../../pages/finder"
+
+const FinderRentalCosts = (props: {
```
Comment (ludtkemorgan) [inline sites/public/src/forms/finder/FinderRentalCosts.tsx:6]: If you want to go down the route where you don't have to pass in all of these props, you could switch to the `useFormContext` style. https://react-hook-form.com/api/useformcontext/. With a fairly small form this isn't necessary though. But if more questions (non multi-select ones) we might want to think to moving to that style

Comment (ColinBuyck) [reply sites/public/src/forms/finder/FinderRentalCosts.tsx:6]: Sounds good! I'll keep this in mind going forward

**Comment by ColinBuyck** [comment]
@ludtkemorgan Those issues should be addressed now 🌞

File: sites/public/pages/finder.tsx [hunk 2]
```diff
@@ -163,58 +208,36 @@ const Finder = () => {
...
-                  <div className="text-3xl pb-4">
-                    {!isDisclaimer
-                      ? t(`finder.${activeQuestion?.fieldGroupName}.question`)
-                      : t("finder.disclaimer.header")}
-                  </div>
-                  <div className="pb-8 border-b border-gray-450">
-                    {!isDisclaimer
-                      ? t("finder.question.subtitle")
-                      : t("finder.disclaimer.subtitle")}
+                  <div className="">
```
Comment (emilyjablonski) [inline sites/public/pages/finder.tsx:211]: 🧹

**Review by ludtkemorgan — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**